DigitalLearningSolutions.Data/ApiClients/LearningHubApiClient.cs

```diff
@@ -103,5 +108,19 @@ private static string GetQueryString(string key, string? value)
         {
             return string.IsNullOrEmpty(value) ? "" : $"{key}={value}";
         }
+
+        private async Task<string?> GetAsyncAndProcessResult(string requestUri)
+        {
+            var response = await client.GetAsync(requestUri);
+            if (response.IsSuccessStatusCode)
+            {
+                return await response.Content.ReadAsStringAsync();
+            }
+
+            var message =
+                $"Learning Hub Open Api call failed. Status code {response.StatusCode.GetHashCode()} ({response.StatusCode}) when trying {requestUri}";

```

Comment:
Not sure you want the hash of the status code.

Reply:
Might also be worth including the response message in case the API sends an error message - https://stackoverflow.com/a/17082328

Reply:
I've refactored the display of these so we grab the value and text in a better fashion.

Are we expecting the API to send its own custom messages for these? Looks like the HttpError class doesn't exist in asp core 3.1 and up, so we'd probably need to make our own class (ideally matching what the API sends back as closely as possible) and use JsonConvert.DeserializeObject as elsewhere to get the message specifically.

Reply:
Ah fair. ReasonPhrase looks good.